### PR TITLE
Sign Up Component

### PR DESCRIPTION
**Pull request by JohnTurkson**

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -4,23 +4,35 @@ import {
...
-                    </Page>
-                </Route>
-                <Route path="/">
-                    <Page>
-                        <Home />
-                    </Page>
-                </Route>
-            </Switch>
-        </Router>
+        <SignUpForm information={signUpInformation} onInformationChange={setSignUpInformation}/>
```
Comment (andykwan1124) [inline client/src/App.tsx:20]: Could you make a `Route` for the sign up page and put `SignUpForm` in there? I think we should still keep the `Router` in here.

File: client/src/App.tsx [hunk 2]
```diff
@@ -4,23 +4,35 @@ import {
     Home,
     Profile
 } from "./pages"
+import SignUpForm, { SignUpInformation } from "./pages/SignUpForm"
```
Comment (andykwan1124) [inline client/src/App.tsx:7]: Did you mean to do something like `import SignUpForm from "./components/SignUp"`? I am getting a `can't find module` error here

File: client/src/App.tsx (hunk 1 above)
Comment (JohnTurkson) [reply client/src/App.tsx:20]: Will revert changes to old App.tsx version

File: client/src/App.tsx (hunk 2 above)
Comment (JohnTurkson) [reply client/src/App.tsx:7]: Will revert changes to old App.tsx version

Comment (JohnTurkson) [reply client/src/App.tsx]: Done.

File: client/src/App.tsx (hunk 1 above)
Comment (JohnTurkson) [reply client/src/App.tsx]: Done.

File: client/src/components/data/SignUpData.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export interface SignUpData {
+    name: string
+    email: string
+    password: string
+}
```
Comment (andykwan1124) [inline client/src/components/data/SignUpData.ts:5]: Is there a need to separate this data into a different interface than the props? Personally I think merging them is fine, but it's up to you

**Review by andykwan1124 — COMMENTED** [review]
Just some comments on the placement of parts of the code.

File: client/src/components/data/SignUpData.ts (hunk 3 above)
Comment (JohnTurkson) [reply client/src/components/data/SignUpData.ts:5]: I was thinking we might need this when making requests to the server later on, ideally the UI state is separate from the model state

File: client/src/components/props/SignUpProps.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { SignUpData } from "../data/SignUpData"
+import { Dispatch, FormEvent, SetStateAction } from "react"
+
+export interface SignUpProps {
+    data: SignUpData
+    onDataChange: Dispatch<SetStateAction<SignUpData>>
+    onSubmit: (event: FormEvent) => void
+}
```
[the remark this answers is not in the document]
Comment (JohnTurkson) [reply client/src/components/props/SignUpProps.ts]: Done

**Review by andykwan1124 — APPROVED** [review]

**Pull request merged**